code/game/machinery/martian_printer.dm

```diff
@@ -12,6 +12,7 @@
 	var/cooldown_duration = 10 MINUTES
 	var/cooldown_time = 0
 	var/cooldown_active = 0
+	var/corpse_check = 0 // if set to 1, you can't enter it after ghosting or otherwise being unable to enter corpse (being sharded or deleted)
```

Comment:
Preserve previous behaviour, please

Reply:
I have absolutely no desire to preserve previous behavior because this functionality is only useful in edge cases and will more often than not just annoy ghosts trying to get into it. Use the cooldown if it's being spammed
Deadharold has also instilled a weird arbitrary deadline on me to get this done before Lamprey is run again and since this would necessitate new instances just to make corpse_check = 0, I am incentivized not to